spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySource.java

```diff
@@ -65,22 +69,90 @@ public Object getProperty(String name) {
-            properties.put(key, item.getValue());
+    /**
+     * Gets settings from Azure/Cache to set as configurations. Updates the cache. </br>
+     * </br>
+     * <b>Note</b>: Doesn't update Feature Management, just stores values in cache. Call
+     * {@code initFeatures} to update Feature Management, but make sure its done in the
+     * last {@code AzureConfigPropertySource}
+     * 
+     * @param propertyCache Cached values to use in store. Also contains values the need
+     * to be refreshed.
+     * @throws IOException
+     */
+    public void initProperties(PropertyCache propertyCache) throws IOException {
+        Date date = new Date();
+        if (propertyCache.getContext(storeName) == null) {
+            propertyCache.addContext(storeName, context);
+            // * for wildcard match
+            QueryOptions queryOptions = new QueryOptions().withKeyNames(context + "*").withLabels(label);
+            List<KeyValueItem> items = source.getKeys(storeName, queryOptions);
+            for (KeyValueItem item : items) {
+                String key = item.getKey().trim().substring(context.length()).replace('/', '.');
+                properties.put(key, item.getValue());
+            }
+            propertyCache.addKeyValuesToCache(items, storeName, date);
+
+            // Reading In Features
+            queryOptions = new QueryOptions().withKeyNames(FEATURE_FLAG_PREFIX + "*").withLabels(label);
+            items = source.getKeys(storeName, queryOptions);
+
+            createFeatureSet(items, propertyCache, date);
+
+        } else {
+            // Using Cached values, first updates cache then sets new properties.
+            if (propertyCache.getRefreshKeys(storeName) != null &&
+                    propertyCache.getRefreshKeys(storeName).size() > 0) {
+                for (String refreshKey : propertyCache.getRefreshKeys(storeName)) {
+                    QueryOptions queryOptions = new QueryOptions().withKeyNames(refreshKey).withLabels(label);
+                    List<KeyValueItem> items = source.getKeys(storeName, queryOptions);
+                    propertyCache.addKeyValuesToCache(items, storeName, date);
+                }
+            }
+
+            for (String key : propertyCache.getKeySet(storeName)) {
+                if (key.startsWith(context)) {
+                    String cachedKey = key.trim().substring(context.length()).replace('/', '.');
+                    properties.put(cachedKey, propertyCache.getCachedValue(key));
+                } else {
+                    List<KeyValueItem> items = new ArrayList<KeyValueItem>();
+                    KeyValueItem item = new KeyValueItem();
+                    item.setKey(key);
+                    item.setValue(propertyCache.getCachedValue(key));
+                    item.setContentType(FEATURE_FLAG_CONTENT_TYPE);
+                    items.add(item);
+
+                    createFeatureSet(items, propertyCache, date);
```

Comment:
The updated features will not go to properties?

Reply:
That is the intent see comment about "distinguish features".